### ISSUE TITLE
NIC private ip is not working

### ISSUE THREAD

**Issue by wallverb** [Azure/azure-xplat-cli#2480]
Running commands:

`azure network nic create ... -a 10.1.2.1 ....` and `azure network nic set ... -a 10.1.2.1 ...` completely ignores -a flag (the same with --private-ip-address)

It might be related to typo in code - when i try to see what `nic = self._parseNic(resourceGroupName, nic, options, _);` returns in `set` method in `nic._js` file then I will get:

```
NetworkInterfaceIPConfiguration {
  id: '/*******',
  name: 'Nic-IP-config',
  etag: 'W/"af6ef8f8-ac93-40d3-82b4-3a93acfca72b"',
  privateIPAddress: '10.1.0.11',
  privateIPAllocationMethod: 'Dynamic',
  subnet: 
   Subnet {
     id: '******' },
  provisioningState: 'Succeeded',
  privateIpAllocationMethod: 'Static',
  privateIpAddress: '10.1.2.1' }
```

Notice `privateIpAddress` vs `privateIPAddress` and `privateIPAllocationMethod` vs `privateIpAllocationMethod`

when i inspect `JSON` that is going out then only `privateIPAddress` and `privateIPAllocationMethod` is being sent in body.

My CLI version: 0.9.14

**Comment by romangromov** [comment]
@wallverb can't repro this issue, working fine both create/set:

```
azure network nic show test-roman-group BugNic
info:    Executing command network nic show
+ Looking up the network interface "BugNic"
data:    Id                              : /subscriptions/2c224e7e-3ef5-431d-a57b-e71f4662e3a6/resourceGroups/test-roman-group/providers/Microsoft.Network/networkInterfaces/BugNic
data:    Name                            : BugNic
data:    Type                            : Microsoft.Network/networkInterfaces
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Enable IP forwarding            : false
data:    IP configurations:
data:      Name                          : Nic-IP-config
data:      Provisioning state            : Succeeded
data:      Private IP address            : 10.1.1.5
data:      Private IP allocation method  : Static
data:      Subnet                        : /subscriptions/2c224e7e-3ef5-431d-a57b-e71f4662e3a6/resourceGroups/test-roman-group/providers/Microsoft.Network/virtualNetworks/VnetForNic/subnets/FirstSubnet
data:
info:    network nic show command OK

azure network nic set test-roman-group BugNic -a "10.1.1.6"
info:    Executing command network nic set
+ Looking up the network interface "BugNic"
+ Updating network interface "BugNic"
data:    Id                              : /subscriptions/2c224e7e-3ef5-431d-a57b-e71f4662e3a6/resourceGroups/test-roman-group/providers/Microsoft.Network/networkInterfaces/BugNic
data:    Name                            : BugNic
data:    Type                            : Microsoft.Network/networkInterfaces
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Enable IP forwarding            : false
data:    IP configurations:
data:      Name                          : Nic-IP-config
data:      Provisioning state            : Succeeded
data:      Private IP address            : 10.1.1.6
data:      Private IP allocation method  : Static
data:      Subnet                        : /subscriptions/2c224e7e-3ef5-431d-a57b-e71f4662e3a6/resourceGroups/test-roman-group/providers/Microsoft.Network/virtualNetworks/VnetForNic/subnets/FirstSubnet
data:
info:    network nic set command OK
```

**Comment by romangromov** [comment]
@wallverb can you please provide step-by-step repro so I can help you? Thanks

**Comment by wallverb** [comment]
@romangromov Here is step by step reproduction - this time I used docker on my OSX to make sure it's fresh start :)


`docker run -it --rm node:latest bash`
`npm install -g azure-cli`

`azure -v`
0.9.14

```
azure config set mode arm
info:    Setting "mode" to value "arm"
info:    Changes saved
```

```
azure login
...
info:    Added subscription Free Trial
info:    Setting subscription "Free Trial" as default
+
info:    login command OK
```

```
GROUP=bug-group
LOCATION=westus
VNET=bug-vnet
SUBNET=bug-subnet



azure group create -l $LOCATION -n $GROUP
info:    Executing command group create
+ Getting resource group bug-group                                             
+ Creating resource group bug-group                                            
info:    Created resource group bug-group
data:    Id:                  /subscriptions/...../resourceGroups/bug-group
data:    Name:                bug-group
data:    Location:            westus
data:    Provisioning State:  Succeeded
data:    Tags: null
data:    
info:    group create command OK



azure network vnet create ${GROUP} ${VNET} ${LOCATION} -a 10.1.0.0/16
info:    Executing command network vnet create
+ Looking up the virtual netwrok "bug-vnet"                                    
+ Creating virtual network "bug-vnet"                                          
data:    Id                              : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet
data:    Name                            : bug-vnet
data:    Type                            : Microsoft.Network/virtualNetworks
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Address prefixes:
data:      10.1.0.0/16
info:    network vnet create command OK


azure network vnet subnet create ${GROUP} ${VNET} ${SUBNET} -a 10.1.0.0/16
info:    Executing command network vnet subnet create
+ Looking up virtual network "bug-vnet"                                        
+ Looking up the subnet "bug-subnet"                                           
+ Creating subnet "bug-subnet"                                                 
data:    Id                              : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet
data:    Name                            : bug-subnet
data:    Provisioning state              : Succeeded
data:    Address prefix                  : 10.1.0.0/16
info:    network vnet subnet create command OK


azure network nic create -g ${GROUP} --subnet-name ${SUBNET} --subnet-vnet-name ${VNET}  -l ${LOCATION} -a 10.1.0.100 -n bug-nic
info:    Executing command network nic create
+ Looking up the subnet "bug-subnet"                                           
+ Looking up the network interface "bug-nic"                                   
+ Creating network interface "bug-nic"                                         
data:    Id                              : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic
data:    Name                            : bug-nic
data:    Type                            : Microsoft.Network/networkInterfaces
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Enable IP forwarding            : false
data:    IP configurations:
data:      Name                          : Nic-IP-config
data:      Provisioning state            : Succeeded
data:      Private IP address            : 10.1.0.4
data:      Private IP allocation method  : Dynamic
data:      Subnet                        : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet
data:    
info:    network nic create command OK


azure network nic set -g ${GROUP} -n bug-nic -a "10.1.0.100"
info:    Executing command network nic set
+ Looking up the network interface "bug-nic"                                   
+ Updating network interface "bug-nic"                                         
data:    Id                              : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic
data:    Name                            : bug-nic
data:    Type                            : Microsoft.Network/networkInterfaces
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Enable IP forwarding            : false
data:    IP configurations:
data:      Name                          : Nic-IP-config
data:      Provisioning state            : Succeeded
data:      Private IP address            : 10.1.0.4
data:      Private IP allocation method  : Dynamic
data:      Subnet                        : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet
data:    
info:    network nic set command OK
```

As you can see both times dynamic IP is used.
Here are the `-vv` results of last 2 commands

```
azure network nic create -g ${GROUP} --subnet-name ${SUBNET} --subnet-vnet-name ${VNET}  -l ${LOCATION} -a 10.1.0.101 -n bug-nic-2 -vv
info:    Executing command network nic create
verbose: Looking up the subnet "bug-subnet"
silly:   requestOptions
silly:   {
silly:       rawResponse: false,
silly:       queryString: {},
silly:       method: 'GET',
silly:       headers: {
silly:           x-ms-client-request-id: '724754a4-6225-4cf2-990a-ada2fd6a422c',
silly:           accept-language: 'en-US',
silly:           Content-Type: 'application/json; charset=utf-8',
silly:           Content-Length: 0,
silly:           Authorization: 'Bearer .......',
silly:           user-agent: 'WindowsAzureXplatCLI/0.9.14'
silly:       },
silly:       url: 'https://management.azure.com/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualnetworks/bug-vnet/subnets/bug-subnet?api-version=2015-06-15',
silly:       body: null
silly:   }
silly:   returnObject
silly:   {
silly:       statusCode: 200,
silly:       header: {
silly:           cache-control: 'no-cache',
silly:           pragma: 'no-cache',
silly:           content-length: '578',
silly:           content-type: 'application/json; charset=utf-8',
silly:           expires: '-1',
silly:           etag: 'W/"4689aa3d-5ecd-45b9-898f-f864f6d1139b"',
silly:           x-ms-request-id: '98f1c144-93a4-450e-a4e3-ca5b9b136022',
silly:           strict-transport-security: 'max-age=31536000; includeSubDomains',
silly:           server: 'Microsoft-HTTPAPI/2.0',
silly:           x-ms-ratelimit-remaining-subscription-reads: '14999',
silly:           x-ms-correlation-request-id: '6822a289-ea33-431b-8eae-d4288cef2278',
silly:           x-ms-routing-request-id: 'EASTUS2:20160128T141906Z:6822a289-ea33-431b-8eae-d4288cef2278',
silly:           date: 'Thu, 28 Jan 2016 14:19:05 GMT',
silly:           connection: 'close'
silly:       },
silly:       body: '{\015\n  "name": "bug-subnet",\015\n  "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet",\015\n  "etag": "W/\\"4689aa3d-5ecd-45b9-898f-f864f6d1139b\\"",\015\n  "properties": {\015\n    "provisioningState": "Succeeded",\015\n    "addressPrefix": "10.1.0.0/16",\015\n    "ipConfigurations": [\015\n      {\015\n        "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic/ipConfigurations/Nic-IP-config"\015\n      }\015\n    ]\015\n  }\015\n}'
silly:   }
verbose: Looking up the network interface "bug-nic-2"
silly:   requestOptions
silly:   {
silly:       rawResponse: false,
silly:       queryString: {},
silly:       method: 'GET',
silly:       headers: {
silly:           x-ms-client-request-id: '32db107d-515e-45fe-8eb4-a5a46087777c',
silly:           accept-language: 'en-US',
silly:           Content-Type: 'application/json; charset=utf-8',
silly:           Content-Length: 0,
silly:           Authorization: 'Bearer .......',
silly:           user-agent: 'WindowsAzureXplatCLI/0.9.14'
silly:       },
silly:       url: 'https://management.azure.com/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2?api-version=2015-06-15',
silly:       body: null
silly:   }
silly:   returnObject
silly:   {
silly:       statusCode: 404,
silly:       header: {
silly:           cache-control: 'no-cache',
silly:           pragma: 'no-cache',
silly:           content-type: 'application/json; charset=utf-8',
silly:           expires: '-1',
silly:           x-ms-failure-cause: 'gateway',
silly:           x-ms-request-id: 'e9ce93e1-17cd-4dfb-9cdd-bf98bdcb5087',
silly:           x-ms-correlation-request-id: 'e9ce93e1-17cd-4dfb-9cdd-bf98bdcb5087',
silly:           x-ms-routing-request-id: 'EASTUS2:20160128T141907Z:e9ce93e1-17cd-4dfb-9cdd-bf98bdcb5087',
silly:           strict-transport-security: 'max-age=31536000; includeSubDomains',
silly:           date: 'Thu, 28 Jan 2016 14:19:07 GMT',
silly:           connection: 'close',
silly:           content-length: '158'
silly:       },
silly:       body: "{"error":{"code":"ResourceNotFound","message":"The Resource 'Microsoft.Network/networkInterfaces/bug-nic-2' under resource group 'bug-group' was not found."}}"
silly:   }
verbose: Creating network interface "bug-nic-2"
silly:   requestOptions
silly:   {
silly:       rawResponse: false,
silly:       queryString: {},
silly:       method: 'PUT',
silly:       headers: {
silly:           x-ms-client-request-id: 'f5f47d8d-41fe-4b53-8f50-4b00a5202def',
silly:           accept-language: 'en-US',
silly:           Content-Type: 'application/json; charset=utf-8',
silly:           Content-Length: 264,
silly:           Authorization: 'Bearer .......',
silly:           user-agent: 'WindowsAzureXplatCLI/0.9.14'
silly:       },
silly:       url: 'https://management.azure.com/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2?api-version=2015-06-15',
silly:       body: '{"location":"westus","properties":{"ipConfigurations":[{"name":"Nic-IP-config","properties":{"subnet":{"id":"/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet"}}}]}}'
silly:   }
silly:   returnObject
silly:   {
silly:       statusCode: 201,
silly:       header: {
silly:           cache-control: 'no-cache',
silly:           pragma: 'no-cache',
silly:           content-length: '1289',
silly:           content-type: 'application/json; charset=utf-8',
silly:           expires: '-1',
silly:           x-ms-request-id: '75087f32-6259-4d2a-9496-3369db30c6e6',
silly:           azure-asyncoperation: 'https://management.azure.com/subscriptions/...../providers/Microsoft.Network/locations/westus/operations/75087f32-6259-4d2a-9496-3369db30c6e6?api-version=2015-06-15',
silly:           strict-transport-security: 'max-age=31536000; includeSubDomains',
silly:           server: 'Microsoft-HTTPAPI/2.0',
silly:           x-ms-ratelimit-remaining-subscription-writes: '1199',
silly:           x-ms-correlation-request-id: '0e9db308-8e20-4c81-a6e1-552b3155a79e',
silly:           x-ms-routing-request-id: 'EASTUS2:20160128T141910Z:0e9db308-8e20-4c81-a6e1-552b3155a79e',
silly:           date: 'Thu, 28 Jan 2016 14:19:09 GMT',
silly:           connection: 'close'
silly:       },
silly:       body: '{\015\n  "name": "bug-nic-2",\015\n  "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2",\015\n  "etag": "W/\\"d136e22e-a3be-4764-bc77-187be814c96b\\"",\015\n  "type": "Microsoft.Network/networkInterfaces",\015\n  "location": "westus",\015\n  "properties": {\015\n    "provisioningState": "Succeeded",\015\n    "resourceGuid": "d6ffe1f7-8741-4dbf-b0c6-85224fa2d6e5",\015\n    "ipConfigurations": [\015\n      {\015\n        "name": "Nic-IP-config",\015\n        "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2/ipConfigurations/Nic-IP-config",\015\n        "etag": "W/\\"d136e22e-a3be-4764-bc77-187be814c96b\\"",\015\n        "properties": {\015\n          "provisioningState": "Succeeded",\015\n          "privateIPAddress": "10.1.0.5",\015\n          "privateIPAllocationMethod": "Dynamic",\015\n          "subnet": {\015\n            "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet"\015\n          },\015\n          "primary": true\015\n        }\015\n      }\015\n    ],\015\n    "dnsSettings": {\015\n      "dnsServers": [],\015\n      "appliedDnsServers": []\015\n    },\015\n    "enableIPForwarding": false\015\n  }\015\n}'
silly:   }
data:    Id                              : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2
data:    Name                            : bug-nic-2
data:    Type                            : Microsoft.Network/networkInterfaces
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Enable IP forwarding            : false
data:    IP configurations:
data:      Name                          : Nic-IP-config
data:      Provisioning state            : Succeeded
data:      Private IP address            : 10.1.0.5
data:      Private IP allocation method  : Dynamic
data:      Subnet                        : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet
data:    
info:    network nic create command OK


azure network nic set -g ${GROUP} -n bug-nic-2 -a "10.1.0.101" -vv
info:    Executing command network nic set
verbose: Looking up the network interface "bug-nic-2"
silly:   requestOptions
silly:   {
silly:       rawResponse: false,
silly:       queryString: {},
silly:       method: 'GET',
silly:       headers: {
silly:           x-ms-client-request-id: '45a2addd-004b-4eeb-859d-c6bb039ebb6a',
silly:           accept-language: 'en-US',
silly:           Content-Type: 'application/json; charset=utf-8',
silly:           Content-Length: 0,
silly:           Authorization: 'Bearer .......',
silly:           user-agent: 'WindowsAzureXplatCLI/0.9.14'
silly:       },
silly:       url: 'https://management.azure.com/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2?api-version=2015-06-15',
silly:       body: null
silly:   }
silly:   returnObject
silly:   {
silly:       statusCode: 200,
silly:       header: {
silly:           cache-control: 'no-cache',
silly:           pragma: 'no-cache',
silly:           content-length: '1289',
silly:           content-type: 'application/json; charset=utf-8',
silly:           expires: '-1',
silly:           etag: 'W/"d136e22e-a3be-4764-bc77-187be814c96b"',
silly:           x-ms-request-id: '79dda166-7174-4dcf-8236-8c9a7466ad8d',
silly:           strict-transport-security: 'max-age=31536000; includeSubDomains',
silly:           server: 'Microsoft-HTTPAPI/2.0',
silly:           x-ms-ratelimit-remaining-subscription-reads: '14996',
silly:           x-ms-correlation-request-id: 'becc0389-afaf-46fa-96de-910ed3b52ba9',
silly:           x-ms-routing-request-id: 'EASTUS2:20160128T141951Z:becc0389-afaf-46fa-96de-910ed3b52ba9',
silly:           date: 'Thu, 28 Jan 2016 14:19:50 GMT',
silly:           connection: 'close'
silly:       },
silly:       body: '{\015\n  "name": "bug-nic-2",\015\n  "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2",\015\n  "etag": "W/\\"d136e22e-a3be-4764-bc77-187be814c96b\\"",\015\n  "type": "Microsoft.Network/networkInterfaces",\015\n  "location": "westus",\015\n  "properties": {\015\n    "provisioningState": "Succeeded",\015\n    "resourceGuid": "d6ffe1f7-8741-4dbf-b0c6-85224fa2d6e5",\015\n    "ipConfigurations": [\015\n      {\015\n        "name": "Nic-IP-config",\015\n        "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2/ipConfigurations/Nic-IP-config",\015\n        "etag": "W/\\"d136e22e-a3be-4764-bc77-187be814c96b\\"",\015\n        "properties": {\015\n          "provisioningState": "Succeeded",\015\n          "privateIPAddress": "10.1.0.5",\015\n          "privateIPAllocationMethod": "Dynamic",\015\n          "subnet": {\015\n            "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet"\015\n          },\015\n          "primary": true\015\n        }\015\n      }\015\n    ],\015\n    "dnsSettings": {\015\n      "dnsServers": [],\015\n      "appliedDnsServers": []\015\n    },\015\n    "enableIPForwarding": false\015\n  }\015\n}'
silly:   }
verbose: Updating network interface "bug-nic-2"
silly:   requestOptions
silly:   {
silly:       rawResponse: false,
silly:       queryString: {},
silly:       method: 'PUT',
silly:       headers: {
silly:           x-ms-client-request-id: '1327e37b-e34f-4ac1-a292-7e933d133431',
silly:           accept-language: 'en-US',
silly:           Content-Type: 'application/json; charset=utf-8',
silly:           Content-Length: 1011,
silly:           Authorization: 'Bearer .......',
silly:           user-agent: 'WindowsAzureXplatCLI/0.9.14'
silly:       },
silly:       url: 'https://management.azure.com/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2?api-version=2015-06-15',
silly:       body: '{"id":"/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2","name":"bug-nic-2","type":"Microsoft.Network/networkInterfaces","location":"westus","etag":"W/\\"d136e22e-a3be-4764-bc77-187be814c96b\\"","properties":{"ipConfigurations":[{"id":"/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2/ipConfigurations/Nic-IP-config","name":"Nic-IP-config","etag":"W/\\"d136e22e-a3be-4764-bc77-187be814c96b\\"","properties":{"privateIPAddress":"10.1.0.5","privateIPAllocationMethod":"Dynamic","subnet":{"id":"/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet"},"provisioningState":"Succeeded"}}],"dnsSettings":{"dnsServers":[],"appliedDnsServers":[]},"enableIPForwarding":false,"resourceGuid":"d6ffe1f7-8741-4dbf-b0c6-85224fa2d6e5","provisioningState":"Succeeded"}}'
silly:   }
silly:   returnObject
silly:   {
silly:       statusCode: 200,
silly:       header: {
silly:           cache-control: 'no-cache',
silly:           pragma: 'no-cache',
silly:           content-length: '1289',
silly:           content-type: 'application/json; charset=utf-8',
silly:           expires: '-1',
silly:           x-ms-request-id: '437b48f4-a266-4478-b9f4-77b5b2a6d7b8',
silly:           azure-asyncoperation: 'https://management.azure.com/subscriptions/...../providers/Microsoft.Network/locations/westus/operations/437b48f4-a266-4478-b9f4-77b5b2a6d7b8?api-version=2015-06-15',
silly:           strict-transport-security: 'max-age=31536000; includeSubDomains',
silly:           server: 'Microsoft-HTTPAPI/2.0',
silly:           x-ms-ratelimit-remaining-subscription-writes: '1198',
silly:           x-ms-correlation-request-id: 'a1170b3e-d7c9-4e87-b227-f26b30675c90',
silly:           x-ms-routing-request-id: 'EASTUS2:20160128T141952Z:a1170b3e-d7c9-4e87-b227-f26b30675c90',
silly:           date: 'Thu, 28 Jan 2016 14:19:51 GMT',
silly:           connection: 'close'
silly:       },
silly:       body: '{\015\n  "name": "bug-nic-2",\015\n  "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2",\015\n  "etag": "W/\\"a20426f2-48d3-44ac-89b0-859f24f7c14f\\"",\015\n  "type": "Microsoft.Network/networkInterfaces",\015\n  "location": "westus",\015\n  "properties": {\015\n    "provisioningState": "Succeeded",\015\n    "resourceGuid": "d6ffe1f7-8741-4dbf-b0c6-85224fa2d6e5",\015\n    "ipConfigurations": [\015\n      {\015\n        "name": "Nic-IP-config",\015\n        "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2/ipConfigurations/Nic-IP-config",\015\n        "etag": "W/\\"a20426f2-48d3-44ac-89b0-859f24f7c14f\\"",\015\n        "properties": {\015\n          "provisioningState": "Succeeded",\015\n          "privateIPAddress": "10.1.0.5",\015\n          "privateIPAllocationMethod": "Dynamic",\015\n          "subnet": {\015\n            "id": "/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet"\015\n          },\015\n          "primary": true\015\n        }\015\n      }\015\n    ],\015\n    "dnsSettings": {\015\n      "dnsServers": [],\015\n      "appliedDnsServers": []\015\n    },\015\n    "enableIPForwarding": false\015\n  }\015\n}'
silly:   }
data:    Id                              : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2
data:    Name                            : bug-nic-2
data:    Type                            : Microsoft.Network/networkInterfaces
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Enable IP forwarding            : false
data:    IP configurations:
data:      Name                          : Nic-IP-config
data:      Provisioning state            : Succeeded
data:      Private IP address            : 10.1.0.5
data:      Private IP allocation method  : Dynamic
data:      Subnet                        : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet
data:    
info:    network nic set command OK
```

As you can see information about private IP never gets into `body` part of requests


You can see that my private ip (passed via -a or --private-ip-address is ignored)


Now if I add `console.log("Parsed options", nic);` in `/usr/local/lib/node_modules/azure-cli/lib/commands/arm/network/nic.js` on line 83 (right after `return self._parseNic(...` in `set` command you will see the `privateIPAddress` vs `privateIpAddress`

```
azure network nic set -g ${GROUP} -n bug-nic-2 -a "10.1.0.101"

+ Looking up the network interface "bug-nic-2"                                 
Parsed options NetworkInterface {
  id: '/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2',
  name: 'bug-nic-2',
  type: 'Microsoft.Network/networkInterfaces',
  location: 'westus',
  etag: 'W/"a20426f2-48d3-44ac-89b0-859f24f7c14f"',
  ipConfigurations: 
   [ NetworkInterfaceIPConfiguration {
       id: '/subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2/ipConfigurations/Nic-IP-config',
       name: 'Nic-IP-config',
       etag: 'W/"a20426f2-48d3-44ac-89b0-859f24f7c14f"',
       privateIPAddress: '10.1.0.5',
       privateIPAllocationMethod: 'Dynamic',
       subnet: [Object],
       provisioningState: 'Succeeded',
       privateIpAllocationMethod: 'Static',
       privateIpAddress: '10.1.0.101' } ],
  dnsSettings: NetworkInterfaceDnsSettings { dnsServers: [], appliedDnsServers: [] },
  enableIPForwarding: false,
  resourceGuid: 'd6ffe1f7-8741-4dbf-b0c6-85224fa2d6e5',
  provisioningState: 'Succeeded' }
+ Updating network interface "bug-nic-2"                                       
data:    Id                              : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/networkInterfaces/bug-nic-2
data:    Name                            : bug-nic-2
data:    Type                            : Microsoft.Network/networkInterfaces
data:    Location                        : westus
data:    Provisioning state              : Succeeded
data:    Enable IP forwarding            : false
data:    IP configurations:
data:      Name                          : Nic-IP-config
data:      Provisioning state            : Succeeded
data:      Private IP address            : 10.1.0.5
data:      Private IP allocation method  : Dynamic
data:      Subnet                        : /subscriptions/...../resourceGroups/bug-group/providers/Microsoft.Network/virtualNetworks/bug-vnet/subnets/bug-subnet
data:    
info:    network nic set command OK
```

If I replace value of `privateIPAddress` with `privateIpAddress` and `privateIPAllocationMethod` with `privateIpAllocationMethod` on that line then it will work fine - so looks like generally Azure CLI ignores that `privateIpAddress`/`privateIpAllocationMethod` (because it probably expects `privateIPAddress`/`privateIPAllocationMethod`)

**Comment by romangromov** [comment]
@wallverb thanks a lot, got your problem. Known issue, already fixed in `dev` branch of this repo, will be shipped with release of `0.9.15`.

Duplicates https://github.com/Azure/azure-xplat-cli/issues/2436, 
PR with fix https://github.com/Azure/azure-xplat-cli/pull/2440 

You can apply fix on your Xplat-CLI local copy or clone this repo and switch to `dev` branch or wait for next `0.9.15`

Hope this helps,
-Roman

**Comment by wallverb** [comment]
@romangromov Awesome - thanks a lot for your help
